endpoints/setuid.go

```diff
@@ -148,6 +163,58 @@ func NewSetUIDEndpoint(cfg *config.Configuration, syncersByBidder map[string]use
 	})
 }
 
+// extractGDPRInfo looks for the GDPR consent string and GDPR signal in the GPP query params
+// first and the 'gdpr' and 'gdpr_consent' query params second.
+func extractGDPRInfo(query url.Values) (gdpr.RequestInfo, error) {
+	// Defualt GDPR values
+	gdprConsent := query.Get("gdpr_consent")
+	gdprSignal := gdpr.SignalAmbiguous
+
+	// Signal from gpp_sid list takes precedence
+	gppSID, err := stringutil.StrToInt8Slice(query.Get("gpp_sid"))
+	if err != nil {
+		return gdpr.RequestInfo{}, fmt.Errorf("Error parsing gpp_sid %s", err.Error())
+	}
+
+	if len(gppSID) > 0 {
+		gdprSignal = gdpr.SignalNo
+		if gppPrivacy.IsSIDInList(gppSID, gppConstants.SectionTCFEU2) {
+			gdprSignal = gdpr.SignalYes
+		}
+	} else {
+		// Signal from gdpr field, if any
+		gdprQuerySignal := query.Get("gdpr")
+		if gdprQuerySignal != "" && gdprQuerySignal != "0" && gdprQuerySignal != "1" {
+			return gdpr.RequestInfo{}, errors.New("the gdpr query param must be either 0 or 1. You gave " + gdprQuerySignal)
+		}
+
+		if i, err := strconv.Atoi(gdprQuerySignal); err == nil {
+			gdprSignal = gdpr.Signal(i)
+		}
+	}
+
+	// Consent from gpp string takes precedence, override gdprConsent in found in "gpp" field
+	if gppString := query.Get("gpp"); len(gppString) > 0 {
+		gpp, err := gpplib.Parse(gppString)
+		if err != nil {
+			return gdpr.RequestInfo{}, err
+		}
+
+		if i := gppPrivacy.IndexOfSID(gpp, gppConstants.SectionTCFEU2); i >= 0 {
+			gdprConsent = gpp.Sections[i].GetValue()
```

Comment:
Discussed offline. The requirements do not appear to be up to date. It was agreed upon that we should always favor the GPP values over the legacy values.

Reply:
An edge case was identified that does not appear to be called out in the spec. If both GPP and the legacy location contain consent strings, we will always favor the GPP string, even if it is malformed. For now we will report the error found in the GPP string rather than falling back to the legacy location. This can always be added later. We can discuss with the committee.